File: easy/2748. Number of Beautiful Pairs/solution.cpp
class Solution {
public:
    int countBeautifulPairs(vector<int>& nums) {
        int count = 0;
        int n = nums.size();
        
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int first = firstDigit(nums[i]);
                int last = nums[j] % 10;
                
                if (gcd(first, last) == 1) {
                    count++;
                }
            }
        }
        
        return count;
    }
    
private:
    int firstDigit(int num) {
        while (num >= 10) {
            num /= 10;
        }
        return num;
    }
    
    int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }
};

File: easy/2748. Number of Beautiful Pairs/solution.py
class Solution:
    def countBeautifulPairs(self, nums: List[int]) -> int:
        def first_digit(num):
            while num >= 10:
                num //= 10
            return num
        
        def gcd(a, b):
            if b == 0:
                return a
            return gcd(b, a % b)
        
        count = 0
        n = len(nums)
        
        for i in range(n):
            for j in range(i + 1, n):
                first = first_digit(nums[i])
                last = nums[j] % 10
                
                if gcd(first, last) == 1:
                    count += 1
        
        return count